verifier/src/main/java/ch/admin/bag/covidcertificate/verifier/modes/ChooseModeDialogFragment.kt

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2021 Ubique Innovation AG <https://www.ubique.ch>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+package ch.admin.bag.covidcertificate.verifier.modes
+
+import android.content.res.ColorStateList
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.graphics.toColorInt
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.activityViewModels
+import ch.admin.bag.covidcertificate.common.config.CheckModeInfoEntry
+import ch.admin.bag.covidcertificate.verifier.R
+import ch.admin.bag.covidcertificate.verifier.databinding.DialogFragmentChooseModeBinding
+import ch.admin.bag.covidcertificate.verifier.databinding.ItemModeButtonBinding
+import ch.admin.bag.covidcertificate.verifier.databinding.ItemModeInfoBinding
+
+class ChooseModeDialogFragment : DialogFragment() {
+
+	companion object {
+		fun newInstance(): ChooseModeDialogFragment = ChooseModeDialogFragment()
+	}
+
+	private var _binding: DialogFragmentChooseModeBinding? = null
+	private val binding get() = _binding!!
+
+	private val viewModel by activityViewModels<ModesAndConfigViewModel>()
+
+	override fun onCreate(savedInstanceState: Bundle?) {
+		super.onCreate(savedInstanceState)
+		setStyle(STYLE_NO_TITLE, R.style.CovidCertificate_ChooseModeDialog)
+	}
+
+	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+		_binding = DialogFragmentChooseModeBinding.inflate(inflater, container, false)
+		return binding.root
+	}
+
+	override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+		viewModel.modesLiveData.observe(viewLifecycleOwner) { modes ->
+
+			binding.modeItemsLayout.removeAllViews()
+
+			for (mode in modes) {
+				val itemView = ItemModeButtonBinding.inflate(layoutInflater, binding.infoItemsLayout, false)
+				itemView.title.text = mode.title
+
+				viewModel.selectedModeLiveData.observe(viewLifecycleOwner) { selectedMode ->
```

Comment:
If `viewModel.modesLiveData` changes its value and this line adds new observers for each mode, the old observers would still be around and trigger when the selected mode changes. Since they access the itemView which might not be added anymore, couldn't this lead to a crash?